core/src/proof.rs

```diff
@@ -100,6 +100,45 @@ pub struct VerifiedPathProof {
     terminal: Option<LeafData>,
 }
 
+impl VerifiedPathProof {
+    /// Get the terminal node.
+    pub fn terminal(&self) -> Option<&LeafData> {
+        self.terminal.as_ref()
+    }
+
+    /// Get the proven path.
+    pub fn path(&self) -> &BitSlice<u8, Msb0> {
+        &self.key_path.view_bits()[..self.depth as usize]
+    }
+
+    /// Check whether this path resolves to the given leaf.
+    ///
+    /// A return value of `true` confirms that the key indeed has no value in the trie.
+    /// A return value of `false` means that either the key provided is out of scope for this path
```

Comment:
review: a few options for "out of scope" queries in this function and `confirm_nonexistence`

1. Just return false (approach used here). pro: simple API, con: yields false negatives, may be confusing
2. Return an option. pro: clear semantics, con: complex API
3. panic. pro: simple API, con: requires checking

thoughts?

Reply:
The error can raise in two cases:

1. somebody is tricking so the execution should not proceed. A `false` is ok in this case.
2. somebody is trying to debug this piece of software or a piece of software that provides proofs. In this case, they would probably appreciate the differentiation between different cases: not a leaf but terminator, leaf but with different key path, leaf but with different value hash.

I am pretty sure that the guys from the second category (maybe us) will appreciate a more detail error description.